lib/state/revision/reducer.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	REVISION_IS_VIEWING,
+	REVISION_SELECT,
+} from '../action-types';
+
+export const isViewingRevisions = ( state = false, { type, isViewing } ) =>
+	REVISION_IS_VIEWING === type ? isViewing : state;
+
+export const selectedRevision = ( state = null, { type, revision } ) =>
+	REVISION_SELECT === type ? revision : state;
+
+export default combineReducers( {
+	isViewingRevisions,
```

Comment:
I feel that's the way to go, but I didn't want to create any conflicts with #402, and still keep the behaviour identical to what it is right now but just replacing the internal state with Redux.